test/models/IllnessStartDateSchema.test.ts

```diff
@@ -0,0 +1,157 @@
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject undefined values', () => {
+        const validationResult = validator.validate({
+            day: undefined,
+            month: undefined,
+            year: undefined,
+            illnessStart: undefined
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject null values', () => {
+        const validationResult = validator.validate({
+            day: null,
+            month: null,
+            year: null,
+            illnessStart: null
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject empty values', () => {
+        const validationResult = validator.validate({
+            day: '',
+            month: '',
+            year: '',
+            illnessStart: ''
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject blank values', () => {
+        const validationResult = validator.validate({
+            day: ' ',
+            month: ' ',
+            year: ' ',
+            illnessStart: ' '
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
+    it('should reject invalid date patterns', () => {
+        const validationResult = validator.validate({
+            day: '001',
+            month: 'abc',
+            year: '*@&',
+            illnessStart: '0'
+        });
+        assertValidationErrors(validationResult, [
+            new ValidationError(dayField, 'You must enter a day'),
+            new ValidationError(monthField, 'You must enter a month'),
+            new ValidationError(yearField, 'You must enter a year'),
+            new ValidationError(startDateField, 'Enter a real date')
+        ]);
+    });
+
```

Comment:
Unnecessary blank line here and in one place below. In turn there is no blank line before last test case.